src/collection.ts

```diff
@@ -493,7 +493,10 @@ export class Collection<TSchema extends Document = Document> {
     filter: Filter<TSchema> = {},
     options: FindOptions = {}
   ): Promise<WithId<TSchema> | null> {
-    return this.find(filter, options).limit(-1).batchSize(1).next();
+    const cursor = this.find(filter, options).limit(-1).batchSize(1);
+    const res = await cursor.next();
+    await cursor.close();
```

Comment:
Calling close without try/finally here I believe is correct, a cursor that receives an error always calls close. Also, calling close unconditionally makes sense because close already includes the logic to no-op when the cursor has already been closed.